api/types/server.go

```diff
@@ -392,12 +394,39 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 	// defaults.ServerAnnounceTTL).
 	s.setStaticFields()
 	if err := s.Metadata.CheckAndSetDefaults(); err != nil {
-		return trace.Wrap(err)
+		// if the server is a registered OpenSSH node, allow the name
+		// to be randomly generated
+		if s.SubKind == KindOpenSSHNode && trace.UserMessage(err) == "missing parameter Name" {
+			s.Metadata.Name = uuid.New().String()
+			if err := s.Metadata.CheckAndSetDefaults(); err != nil {
+				return trace.Wrap(err)
+			}
+		} else {
+			return trace.Wrap(err)
+		}
 	}
 
 	if s.Kind == "" {
 		return trace.BadParameter("server Kind is empty")
 	}
+	if s.SubKind == "" {
+		s.SubKind = KindTeleportNode
+	} else if s.SubKind == KindOpenSSHNode {
+		if s.Spec.Addr == "" {
+			return trace.BadParameter(`Addr must be set when server SubKind is "openssh"`)
+		}
+		if s.Spec.PublicAddr != "" {
+			return trace.BadParameter(`PublicAddr must not be set when server SubKind is "openssh"`)
+		}
+		if s.Spec.Hostname == "" {
+			return trace.BadParameter(`Hostname must be set when server SubKind is "openssh"`)
+		}
+
+		_, err := netip.ParseAddrPort(s.Spec.Addr)
+		if err != nil {
+			return trace.BadParameter("invalid Addr: %v", err)
```

Comment:
Include `s.Spec.Addr` in the error message.

Reply:
Done